frame/vamm/src/tests.rs

```diff
@@ -0,0 +1,181 @@
+	fn create_vamm_zero_peg_multiplier_error(base_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, quote_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, peg_multiplier in  ZERO_RESERVE..=ZERO_RESERVE) {
+		ExtBuilder::default().build().execute_with(|| {
+			assert_noop!(Vamm::create(base_asset_reserves, quote_asset_reserves, peg_multiplier), Error::<MockRuntime>::PegMultiplierIsZero);
+		})
+	}
+}
+
+proptest! {
+	#![proptest_config(ProptestConfig::with_cases(RUN_CASES))]
+	#[test]
+	fn create_vamm_update_counter_succeeds(base_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, quote_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, peg_multiplier in MINIMUM_RESERVE..=MAXIMUM_RESERVE, loop_times in MINIMUM_RESERVE..=100) {
+		ExtBuilder::default().build().execute_with(|| {
+			let markets = Vamm::vamm_count();
+
+			for _ in 0..loop_times {
+				assert_ok!(Vamm::create(base_asset_reserves, quote_asset_reserves, peg_multiplier));
+			}
+
+			assert_eq!(Vamm::vamm_count(), markets + loop_times);
+		});
+	}
+}
+
+proptest! {
+	#![proptest_config(ProptestConfig::with_cases(RUN_CASES))]
+	#[test]
+	#[allow(clippy::disallowed_methods)]
+	fn create_vamm_emits_event_succeeds(base_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, quote_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, peg_multiplier in MINIMUM_RESERVE..=MAXIMUM_RESERVE) {
+		ExtBuilder::default().build().execute_with(|| {
+			System::set_block_number(1);
+
+			let vamm_created_ok = Vamm::create(base_asset_reserves, quote_asset_reserves, peg_multiplier);
+			let vamm_created = Vamm::get_vamm(vamm_created_ok.unwrap()).unwrap();
+			assert_ok!(vamm_created_ok);
+
+			System::assert_last_event(Event::Vamm(
+				pallet::Event::Created { vamm_id: 0u128, state: vamm_created}
+			))
+		});
+	}
+}
+
+proptest! {
+	#![proptest_config(ProptestConfig::with_cases(RUN_CASES))]
+	#[test]
+	fn create_vamm_updates_storage_map(base_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, quote_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, peg_multiplier in MINIMUM_RESERVE..=MAXIMUM_RESERVE) {
+		ExtBuilder::default().build().execute_with(|| {
+			assert!(!VammMap::<MockRuntime>::contains_key(0u128));
+
+			let vamm_created_ok = Vamm::create(base_asset_reserves, quote_asset_reserves, peg_multiplier);
+			assert_ok!(vamm_created_ok);
+
+			assert!(VammMap::<MockRuntime>::contains_key(0u128));
+		});
+	}
+}
+
+proptest! {
+	#![proptest_config(ProptestConfig::with_cases(RUN_CASES))]
+	#[test]
+	fn create_vamm_updates_storage_counter(base_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, quote_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, peg_multiplier in MINIMUM_RESERVE..=MAXIMUM_RESERVE) {
+		ExtBuilder::default().build().execute_with(|| {
+			assert_eq!(Vamm::vamm_count(), 0u128);
+
+			let vamm_created_ok = Vamm::create(base_asset_reserves, quote_asset_reserves, peg_multiplier);
+			assert_ok!(vamm_created_ok);
+
+			assert_eq!(Vamm::vamm_count(), 1u128);
+		});
+	}
+}
```

Comment:
You can have all proptests under the same proptest! block. I try to include all proptests related to the same set of functionality (creation, depositing, ...) all under the same proptest! block. Also sometimes you want to run some proptests more or less than others, so they can be under a seperate proptest! block that sets a smaller/larger #![proptest_config(ProptestConfig::with_cases(RUN_CASES))] value

Reply:
Oh, I thought I could only have one function per block. (might be confusing it with propcompose?)

Thank you for the input, there was no need to have different proptest! blocks since all of them was using the same number of test cases.